tiledb/sm/query/query_condition.cc

```diff
@@ -652,23 +652,35 @@ void QueryCondition::apply_ast_node(
         }
       }
     } else {
-      const auto f = result_tile->frag_idx();
+      if (result_tile == nullptr) {
+        throw QueryConditionException(
+            "Failed to apply query condition due to null '" + field_name +
+            "' ResultTile.");
+      }
+
+      const auto& frag_meta = fragment_metadata[result_tile->frag_idx()];
+      // If the tile_tuple is null it was skipped by this fragment in
+      // ReaderBase::skip_field. This is the case for evolved attributes / dims.
+      const auto tile_tuple = result_tile->tile_tuple(field_name);
+      // Special attributes handle null tile_tuples specifically.
+      // We can skip null tile_tuples only if they are not a special attribute.
+      bool skip_tile_tuple = !ArraySchema::is_special_attribute(field_name) &&
+                             tile_tuple == nullptr;
 
       // For delete timestamps conditions, if there's no delete metadata or
       // delete condition was already processed, GT condition is always true.
       if (field_name == constants::delete_timestamps &&
-          (!fragment_metadata[f]->has_delete_meta() ||
-           fragment_metadata[f]
-                   ->loaded_metadata()
-                   ->get_processed_conditions_set()
-                   .count(condition_marker_) != 0)) {
+          (!frag_meta->has_delete_meta() ||
+           frag_meta->loaded_metadata()->get_processed_conditions_set().count(
+               condition_marker_) != 0)) {
         assert(Op == QueryConditionOp::GT);
         for (size_t c = starting_index; c < starting_index + length; ++c) {
           result_cell_bitmap[c] = 1;
         }
         starting_index += length;
         continue;
-      } else if (!fragment_metadata[f]->array_schema()->is_field(field_name)) {
+      } else if (
+          !frag_meta->array_schema()->is_field(field_name) || skip_tile_tuple) {
```

Comment:
Is `is_field(field_name)` always false if `skip_tile_tuple` is true?
Seems like it would be right?  If `skip_tile_tuple` is true then my inference is that it means either (1) the result tile has no field for the given name, or (2) the result tile skipped the field due to evolution.

I'm wondering if removing the first clause here might actually add some clarity here but obviously it will not if it is a necessary clause

Reply:
This crossed my mind yesterday too, but for the test I wrote the `field_name` happens to use the same name as the attribute in the previous schemas. So when we check any previous schema from FragmentMetadata prior to the latest evolution we do have an attribute named `a` but `tile_tuple` is also null because it's been evolved and we skipped initializing the pre-evolution TileTuple via `ReaderBase::skip_field`.